scalar/time.go

```diff
@@ -0,0 +1,71 @@
+package scalar
+
+import (
+	"github.com/apache/arrow/go/v13/arrow"
+)
+
+type Time struct {
```

Comment:
this also needs `Value` func, something like
```go
func (s *Time) Value() any {
	if !s.Valid {
		return nil
	}
	switch width := s.getBitWidth(); width {
	case 64:
		return arrow.Time64(s.Int.Value())
	case 32:
		return arrow.Time32(s.Int.Value())
	default:
		panic("unsupported bit width " + strconv.Itoa(width))
	}
}
```

Reply:
added Get()